components/brave_wallet/browser/brave_wallet_p3a.cc

```diff
@@ -75,21 +88,39 @@ void RecordDefaultSolanaWalletSetting(PrefService* pref_service) {
 
 BraveWalletP3A::BraveWalletP3A(BraveWalletService* wallet_service,
                                KeyringService* keyring_service,
-                               PrefService* pref_service)
+                               PrefService* profile_prefs,
+                               PrefService* local_state)
     : wallet_service_(wallet_service),
       keyring_service_(keyring_service),
-      pref_service_(pref_service) {
+      profile_prefs_(profile_prefs),
+      local_state_(local_state) {
+  DCHECK(profile_prefs);
+  DCHECK(local_state);
+
+  MigrateUsageProfilePrefsToLocalState();
+
   RecordInitialBraveWalletP3AState();
   AddObservers();
+
+  pref_change_registrar_.Init(local_state_);
+  pref_change_registrar_.Add(kBraveWalletLastUnlockTime,
+                             base::BindRepeating(&BraveWalletP3A::ReportUsage,
+                                                 base::Unretained(this), true));
 }
 
-BraveWalletP3A::~BraveWalletP3A() = default;
+BraveWalletP3A::~BraveWalletP3A() {
+  pref_change_registrar_.Remove(kBraveWalletLastUnlockTime);
```

Comment:
https://source.chromium.org/chromium/chromium/src/+/main:components/prefs/pref_change_registrar.h;l=20-21;bpv=1;bpt=1?q=PrefChangeRegistrar&ss=chromium says there is no need to Remove manually